yammer-extra/src/main/java/com/arpnetworking/metrics/yammer/MetricsRegistry.java

```diff
@@ -0,0 +1,226 @@
+/**
+ * Copyright 2015 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.yammer;
+
+import com.arpnetworking.metrics.Metrics;
+import com.arpnetworking.metrics.MetricsFactory;
+import com.arpnetworking.metrics.impl.TsdMetricsFactory;
+import com.arpnetworking.metrics.impl.TsdQueryLogSink;
+import com.yammer.metrics.core.Clock;
+import com.yammer.metrics.core.Gauge;
+import com.yammer.metrics.core.Metric;
+import com.yammer.metrics.core.MetricName;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.Function;
+import javax.annotation.Nonnull;
+
+/**
+ * The replacement class for the MetricRegistry.  In the case of using the shaded library
+ * this class will serve as the provided MetricRegistry.  In the case of using the
+ * supplementary library, you should use this class instead of the DropWizard-provided
+ * Metric registry.
+ *
+ * @author Brandon Arp (barp at groupon dot com)
+ */
+@SuppressFBWarnings("NM_SAME_SIMPLE_NAME_AS_SUPERCLASS")
+public class MetricsRegistry extends com.yammer.metrics.core.MetricsRegistry {
+
+    /**
+     * Public constructor.
+     */
+    public MetricsRegistry() {
+        // TODO(barp): Read the settings from a config file [AINT-639]
```

Comment:
Please move this into an Issue in the ArpNetworking metrics-java-client project and adjust reference. You can link the AINT Jira to the ArpNetworking issue.